app/tournamentBusiness/tournamentProgress.js

```diff
@@ -0,0 +1,121 @@
+import $ from "jquery";
+import * as tournamentAlerts from "./tournamentAlerts";
+
+var tournament;
+
+export function init(t) {
+    tournament = t;
+
+    generateHTMLTournamentBracket();
+}
+
+function generateHTMLTournamentBracket() {
+    cleanBracket();
+    divideBracketVerticallyInRounds(tournament.getRounds());
+    appendAllMatchesWithSpacesToHtml(tournament.matches);
+    bindClickEventOnWinButtons(tournament.matches);
+}
+
+function cleanBracket() {
+    $("#tournamentBracket").empty();
+}
+
+function divideBracketVerticallyInRounds(rounds) {
+    var colSize = 100/rounds;
+
+    for (var r = 1; r <= rounds; r++) {
+        $("#tournamentBracket").append(
+            `<div class="col-xs-1" id="round${r}" style="width: ${colSize}%; padding: 0;"></div>`
+        );
+    }
+}
+
+function appendAllMatchesWithSpacesToHtml() {
+    tournament.matches.forEach( function(match, i, matches) {
+        if (match.round > 1) {
+            appendTransparenSpacesToHtml(match.round, matches[i-1].round);
```

Comment:
This is very fragile because it's based upon the order of the matches. It will fail if matches are reversed, which at the model level would be legit. Try to understand well what's the public contract and what are implementation details you should not base on.

It's also a bad smell to use forEach and then access previous match. You could use a local state variable, but then I would not use forEach() but for loop, because you are not using a pure method.